File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-4.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        length = len(nums)
        l, r = 0, length - 1
        while (l < r):
            m = (l + r) // 2
            if (nums[m] > nums[r]):
                l = m + 1
            else:
                r = m
        pivot = l

        if target <= nums[-1]:
            l, r = pivot, length -1
        else:
            l, r = 0, (pivot - 1 + length) % length
        while (l <= r):
            m = (l + r) // 2
            if (nums[m] == target):
                return m
            elif (nums[m] < target):
                l = m + 1
            else:
                r = m - 1
        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-4.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numberList = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (numberList.containsKey(target - nums[i])) {
                return new int[]{numberList.get(target - nums[i]), i};
            }
            if (numberList.containsKey(nums[i])) {
                continue;
            }
            numberList.put(nums[i], i);
        }

        return new int[]{};
    }
}
